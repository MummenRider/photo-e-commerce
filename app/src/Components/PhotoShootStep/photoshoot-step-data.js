export const steps = [
  {
    id: 1,
    content: "LET'S TALK ABOUT THE PHOTOSHOOT",
  },
  {
    id: 2,
    content: "RESERVE YOUR DATE",
  },
  {
    id: 3,
    content: "YOUR OUTFIT OF THE DAY",
  },
  {
    id: 4,
    content: "LOREM IPSUM DOLOR AMET, CONSETETUR",
  },
  {
    id: 5,
    content: "AT VERO EOS ET ACCUSAM ET JUSTO DUO",
  },
];
