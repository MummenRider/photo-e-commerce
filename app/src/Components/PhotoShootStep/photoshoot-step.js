import {
  Container,
  Left,
  Inner,
  Right,
  Title,
  SubTitle,
  List,
  Item,
  Image,
  ContactBtn,
  SvgInner,
  SvgWave,
  Outer,
  Wrapper,
  Blob,
  BlobsContainer,
} from 'Components/PhotoShootStep/photoshoot-step.styles';
import { useInView } from 'react-intersection-observer';

export default function PhotoShootStep({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

PhotoShootStep.Left = function PhotoShootLeft({ children, ...restProps }) {
  return (
    <Left {...restProps}>
      <Wrapper>{children}</Wrapper>
    </Left>
  );
};

PhotoShootStep.Inner = function PhotoShootInner({ children, ...restProps }) {
  return <Inner {...restProps}>{children}</Inner>;
};

PhotoShootStep.Outer = function PhotoShootOuter({ children, ...restProps }) {
  return <Outer {...restProps}>{children}</Outer>;
};

PhotoShootStep.Right = function PhotoShootRight({
  isMobile,
  children,
  ...restProps
}) {
  const [ref, inView] = useInView();
  return isMobile ? (
    <Right
      style={{ transformOrigin: 'top center' }}
      transition={{
        ease: [0.43, 0.13, 0.23, 0.96],
        duration: 0.8,
      }}
      animate={{
        scale: inView ? 1 : 1.2,
      }}
      {...restProps}
      ref={ref}
    >
      {children}
    </Right>
  ) : null;
};

PhotoShootStep.Title = function PhotoShootTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

PhotoShootStep.SubTitle = function PhotoShootSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

PhotoShootStep.List = function PhotoShootList({ children, ...restProps }) {
  return <List {...restProps}>{children}</List>;
};

PhotoShootStep.Item = function PhotoShootItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

PhotoShootStep.Image = function PhotoShootItem({ ...restProps }) {
  return <Image {...restProps} />;
};

PhotoShootStep.ContactBtn = function PhotoShootContactBtn({
  children,
  ...restProps
}) {
  return (
    <ContactBtn whileHover={{}} {...restProps}>
      {children}
    </ContactBtn>
  );
};

PhotoShootStep.SvgWave = function PhotoShootStepSvgWave({
  children,
  ...restProps
}) {
  return <SvgWave {...restProps}>{children}</SvgWave>;
};

PhotoShootStep.SvgInner = function PhotoShootStepSvgInner({
  children,
  ...restProps
}) {
  return <SvgInner {...restProps}>{children}</SvgInner>;
};

PhotoShootStep.Blob = function PhotoShootStepBlob({ ...restProps }) {
  return <Blob {...restProps} />;
};

PhotoShootStep.BlobsContainer = function PhotoShootStepBlobsContainer({
  children,
  ...restProps
}) {
  return <BlobsContainer {...restProps}>{children}</BlobsContainer>;
};
