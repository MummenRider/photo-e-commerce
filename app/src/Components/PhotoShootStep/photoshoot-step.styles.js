import styled from 'styled-components';
import { motion } from 'framer-motion';
export const Container = styled.section`
  display: flex;
  flex-direction: column;
  background: rgb(240, 238, 236);
  padding: 14vh 0 8vh 0;
  position: relative;
  flex-wrap: wrap;
  &::before {
    content: '';
    display: ${(props) => (props.isMobile ? 'block' : 'none')};
    position: absolute;
    top: -1%;
    left: 50%;
    transform: translate(-50%, 1%);
    width: 0;
    height: 0;
    border-top: 6vh solid #dbd8c7;
    border-left: 3vh solid transparent;
    border-right: 3vh solid transparent;
  }

  & > * {
    flex: 1 0 50%;
  }

  @media screen and (min-width: 768px) {
    flex-direction: row;
  }
`;

export const Left = styled.div`
  color: black;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Right = styled(motion.div)``;
export const Image = styled.div`
  background-image: url(${(props) => props.src});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  width: 100%;
  padding-top: 125%;
`;
export const Inner = styled.div`
  display: flex;
  flex-direction: column;
  align-content: left;

  .goo {
  }
`;

export const Outer = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
`;

export const Title = styled.h1`
  font-family: var(--ff-secondary);
  font-size: 4.2rem;
  font-weight: lighter;
  text-align: left;
  line-height: 1.2;
  padding: 5rem 0 1rem 0;

  @media screen and (min-width: 768px) {
    font-size: clamp(3.2rem, 5vw, 32rem);
    padding: 0;
  }
`;

export const SubTitle = styled.p`
  font-family: var(--ff-primary);
  font-size: clamp(1.5rem, 1.2vw, 7.8rem);
  margin-bottom: 10%;
  span {
    color: #9a6130;
    font-weight: 600;
  }
`;

export const List = styled.ul`
  font-family: var(--ff-primary);
  counter-reset: counterName;
`;

export const Item = styled.p`
  font-size: clamp(1.2rem, 0.8vw, 5.2rem);
  font-weight: 200;
  margin: 2em 0;
  position: relative;
  padding-left: 15%;

  &::before {
    counter-increment: counterName;
    content: '0' counter(counterName) '. ';
    font-size: clamp(1rem, 0.8vw, 4.5rem);
    position: absolute;
    left: 0;
  }
`;

export const Wrapper = styled.div``;

export const ContactBtn = styled(motion.div)`
  border: 3px solid #9a6130;
  text-align: center;
  font-family: var(--ff-primary);
  font-size: clamp(1rem, 1vw, 6rem);
  width: 40%;
  min-width: 100px;
  padding: 1.5em 0;
  position: relative;
  transition: all 700ms ease;
  margin-top: 5%;
  margin-left: 15%;
  z-index: 1;
  color: #9a6130;
  border-radius: 0.2em;
  &::before {
    position: absolute;
    pointer-events: none;
    left: 120%;
    font-size: clamp(0.9rem, 0.5vw, 5rem);
    width: 100%;
    text-align: left;
    line-height: 1.5;
    letter-spacing: 0.5px;
    font-weight: 200;
    color: black;
    width: 80%;
    min-width: 100px;
    content: 'Lorem ipsum dolor sit amet,
consetetu';
  }
  &:hover {
    color: #fff;
    .c-button__blobs {
      div {
        transform: scale(1.4) translateY(0) translateZ(0);
      }
    }
  }
  .gooeyE {
    position: absolute;
    width: 100%;
    height: 100%;
    overflow: hidden;
    top: 0;
    left: 0;

    z-index: -1;
    div {
      pointer-events: none;
      position: absolute;
      height: 100%;
      width: 38%;
      border-radius: 100%;
      background-color: #9a6130;
      transform: scale(1.8) translateY(126%) translateZ(0);
      transition: all 700ms ease;
      &:nth-child(1) {
        left: -5%;
      }
      &:nth-child(2) {
        left: 30%;
        transition-delay: 90ms;
      }
      &:nth-child(3) {
        left: 66%;
        transition-delay: 25ms;
      }
    }
  }
  /* cursor: pointer;
  background-color: none;
  text-align: center;
  font-family: var(--ff-primary);
  font-size: clamp(1rem, 1vw, 6rem);
  font-weight: 500;
  border: 3px solid #9a6130;
  border-radius: 4px;
  padding: 1.4em 1em;
  width: 40%;
  min-width: 100px;
  position: relative;
  margin-top: 5%;
  margin-left: 15%;
  color: #9a6130;

  &::before {
    position: absolute;
    pointer-events: none;
    left: 120%;
    font-size: clamp(0.9rem, 0.5vw, 5rem);
    width: 100%;
    text-align: left;
    line-height: 1.5;
    letter-spacing: 0.5px;
    font-weight: 200;
    color: black;
    width: 80%;
    min-width: 100px;
    content: 'Lorem ipsum dolor sit amet,
consetetu';
  }

  &:hover {
    color: #fff;
    .c-button__blobs {
      div {
        transform: scale(1.4) translateY(0) translateZ(0);
      }
    }
  }
  .gooeyE {
    position: relative;
    border: 1px solid red;
    div {
      pointer-events: none;
      position: absolute;
      height: 4em;
      width: 40%;
      border-radius: 50%;
      background-color: pink;
     
    }
  } */
`;

export const BlobsContainer = styled(motion.div)``;

export const Blob = styled(motion.div)``;

export const SvgInner = styled.div`
  position: absolute;
  bottom: -0.15%;
  left: 0;
  width: 100%;
  overflow: hidden;
  line-height: 0;
  transform: rotate(180deg);

  svg {
    position: relative;
    display: block;
    width: calc(100% + 1.3px);
    height: 15.36vh;
  }

  path {
    fill: #111112;
  }
`;
export const SvgWave = styled.div`
  margin-top: 15em;
`;
