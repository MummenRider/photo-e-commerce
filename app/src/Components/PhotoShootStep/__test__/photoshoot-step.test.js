import PhotoShootStep from '../photoshoot-step';
import { render, cleanup } from '@testing-library/react';
import React from 'react';
import photoshootImg from 'Assets/Images/photoshoot_img.jpg';
import { steps } from 'Components/PhotoShootStep/photoshoot-step-data';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

afterEach(cleanup);
describe('<PhotoShootStep />', () => {
  it('should render <PhotoShootStep /> with data populated', () => {
    const { container, getByText } = render(
      <PhotoShootStep>
        <PhotoShootStep.Left>
          <PhotoShootStep.Outer>
            <PhotoShootStep.Title>This block is a title</PhotoShootStep.Title>
            <PhotoShootStep.SubTitle>Subtitle here</PhotoShootStep.SubTitle>
          </PhotoShootStep.Outer>
          <PhotoShootStep.Inner>
            <PhotoShootStep.List>
              {steps.map((step) => (
                <PhotoShootStep.Item key={step.id}>
                  {step.content}
                </PhotoShootStep.Item>
              ))}
            </PhotoShootStep.List>
            <PhotoShootStep.ContactBtn>Contact me</PhotoShootStep.ContactBtn>
          </PhotoShootStep.Inner>
        </PhotoShootStep.Left>

        <PhotoShootStep.Right isMobile={true}>
          <PhotoShootStep.Image src={photoshootImg} />
        </PhotoShootStep.Right>

        <PhotoShootStep.SvgWave>
          <PhotoShootStep.SvgInner>Svg content here</PhotoShootStep.SvgInner>
        </PhotoShootStep.SvgWave>
      </PhotoShootStep>
    );

    expect(getByText('This block is a title')).toBeTruthy();
    expect(getByText('Subtitle here')).toBeTruthy();
    expect(getByText('RESERVE YOUR DATE')).toBeTruthy();
    expect(getByText('Svg content here')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render <PhotoShootStep /> with data populated', () => {
    const { container, queryByText } = render(
      <PhotoShootStep>
        <PhotoShootStep.Right isMobile={false}>
          I am hidden
        </PhotoShootStep.Right>
      </PhotoShootStep>
    );

    expect(queryByText('I am hidden')).toBeFalsy();
    expect(container).toMatchSnapshot();
  });

  it('should not render the pseudo element (upside down triangle) in smaller device', () => {
    const tree = renderer.create(<PhotoShootStep isMobile={false} />).toJSON();

    expect(tree).toHaveStyleRule('display', 'none', {
      modifier: '::before',
    });
  });

  it('should render the pseudo element (upside down triangle) in bigger device', () => {
    const tree = renderer.create(<PhotoShootStep isMobile={true} />).toJSON();

    expect(tree).toHaveStyleRule('display', 'block', {
      modifier: '::before',
    });
  });
});
