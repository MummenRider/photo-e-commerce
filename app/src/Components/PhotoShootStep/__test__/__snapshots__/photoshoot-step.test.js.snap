// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoShootStep /> should render <PhotoShootStep /> with data populated 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: rgb(240,238,236);
  padding: 14vh 0 8vh 0;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0::before {
  content: "";
  display: none;
  position: absolute;
  top: -1%;
  left: 50%;
  -webkit-transform: translate(-50%,1%);
  -ms-transform: translate(-50%,1%);
  transform: translate(-50%,1%);
  width: 0;
  height: 0;
  border-top: 6vh solid #dbd8c7;
  border-left: 3vh solid transparent;
  border-right: 3vh solid transparent;
}

.c0 > * {
  -webkit-flex: 1 0 50%;
  -ms-flex: 1 0 50%;
  flex: 1 0 50%;
}

.c1 {
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  background-image: url(photoshoot_img.jpeg);
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  width: 100%;
  padding-top: 125%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: left;
  -ms-flex-line-pack: left;
  align-content: left;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  font-family: var(--ff-secondary);
  font-size: 4.2rem;
  font-weight: lighter;
  text-align: left;
  line-height: 1.2;
  padding: 5rem 0 1rem 0;
}

.c4 {
  font-family: var(--ff-primary);
  font-size: clamp(1.5rem,1.2vw,7.8rem);
  margin-bottom: 10%;
}

.c4 span {
  color: #9a6130;
  font-weight: 600;
}

.c6 {
  font-family: var(--ff-primary);
  counter-reset: counterName;
}

.c7 {
  font-size: clamp(1.2rem,0.8vw,5.2rem);
  font-weight: 200;
  margin: 2em 0;
  position: relative;
  left: 10%;
  padding-left: 2em;
}

.c7::before {
  counter-increment: counterName;
  content: "0" counter(counterName) ". ";
  font-size: clamp(1rem,0.8vw,4.5rem);
  position: absolute;
  left: -10%;
}

.c8 {
  cursor: pointer;
  background-color: #9a6130;
  border: none;
  font-family: var(--ff-primary);
  font-size: clamp(1rem,1vw,6rem);
  font-weight: 400;
  padding: 1.4em 1em;
  width: 40%;
  min-width: 100px;
  position: relative;
  margin-top: 5%;
  margin-left: 4.5em;
  box-shadow: rgba(17,17,26,0.33) 0px 4px 16px, rgba(17,17,26,0) 0px 8px 32px;
}

.c8::before {
  position: absolute;
  left: 120%;
  font-size: clamp(0.9rem,0.5vw,5rem);
  width: 100%;
  text-align: left;
  line-height: 1.5;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: 200;
  width: 80%;
  min-width: 100px;
  content: "Lorem ipsum dolor sit amet,consetetu";
}

.c11 {
  position: absolute;
  bottom: -0.5%;
  left: 0;
  width: 100%;
  overflow: hidden;
  line-height: 0;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c11 svg {
  position: relative;
  display: block;
  width: calc(100% + 1.3px);
  height: 15.36vh;
}

.c11 path {
  fill: #111112;
}

.c10 {
  margin-top: 15em;
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    font-size: clamp(3.2rem,5vw,32rem);
    padding: 0;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class=""
      >
        <div
          class="c2"
        >
          <h1
            class="c3"
          >
            This block is a title
          </h1>
          <p
            class="c4"
          >
            Subtitle here
          </p>
        </div>
        <div
          class="c5"
        >
          <ul
            class="c6"
          >
            <p
              class="c7"
            >
              LET'S TALK ABOUT THE PHOTOSHOOT
            </p>
            <p
              class="c7"
            >
              RESERVE YOUR DATE
            </p>
            <p
              class="c7"
            >
              YOUR OUTFIT OF THE DAY
            </p>
            <p
              class="c7"
            >
              LOREM IPSUM DOLOR AMET, CONSETETUR
            </p>
            <p
              class="c7"
            >
              AT VERO EOS ET ACCUSAM ET JUSTO DUO
            </p>
          </ul>
          <button
            class="c8"
          >
            Contact me
          </button>
        </div>
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="c9"
        src="photoshoot_img.jpeg"
      />
    </div>
    <div
      class="c10"
    >
      <div
        class="c11"
      >
        Svg content here
      </div>
    </div>
  </section>
</div>
`;

exports[`<PhotoShootStep /> should render <PhotoShootStep /> with data populated 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: rgb(240,238,236);
  padding: 14vh 0 8vh 0;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0::before {
  content: "";
  display: none;
  position: absolute;
  top: -1%;
  left: 50%;
  -webkit-transform: translate(-50%,1%);
  -ms-transform: translate(-50%,1%);
  transform: translate(-50%,1%);
  width: 0;
  height: 0;
  border-top: 6vh solid #dbd8c7;
  border-left: 3vh solid transparent;
  border-right: 3vh solid transparent;
}

.c0 > * {
  -webkit-flex: 1 0 50%;
  -ms-flex: 1 0 50%;
  flex: 1 0 50%;
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <section
    class="c0"
  />
</div>
`;
